src/cluster_resources.rs

```diff
@@ -0,0 +1,393 @@
+//! A structure containing the cluster resources.
+
+use std::{
+    collections::{BTreeMap, HashSet},
+    fmt::Debug,
+};
+
+use crate::{
+    client::Client,
+    error::{Error, OperatorResult},
+    k8s_openapi::{
+        api::{
+            apps::v1::{DaemonSet, StatefulSet},
+            core::v1::{ConfigMap, ObjectReference, Service},
+        },
+        apimachinery::pkg::apis::meta::v1::{LabelSelector, LabelSelectorRequirement},
+    },
+    kube::{Resource, ResourceExt},
+    labels::{APP_INSTANCE_LABEL, APP_MANAGED_BY_LABEL, APP_NAME_LABEL},
+};
+use kube::core::ErrorResponse;
+use serde::{de::DeserializeOwned, Serialize};
+use tracing::{debug, info};
+
+/// A cluster resource handled by `ClusterResources`.
+pub trait ClusterResource:
+    Clone + Debug + DeserializeOwned + Resource<DynamicType = ()> + Serialize
+{
+}
+
+impl ClusterResource for ConfigMap {}
```

Comment:
Can we maybe add a comment here about this mainly being a marker trait, but that additional implementation is required in the finalize fn if more structs are added here, just to be on the safe side?

Reply:
Comment added. Please have a look!

Reply:
Looks good, thanks!